src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,646 @@
+   * and converts all spaces within that range to snow.
+   */
+  private static class NixAbility extends CommanderAbility
+  {
+    private static final String NIX_NAME = "Nix";
+    private static final int NIX_COST = 1;
+    private static final int NIX_BUFF = 10; // Standard 10
+
+    private int numActivations = 0;
+
+    CommanderAve Ave;
+    COModifier damageMod = null;
+
+    NixAbility(CommanderAve commander)
+    {
+      super(commander, NIX_NAME, NIX_COST);
+      Ave = commander;
+      damageMod = new CODamageModifier(NIX_BUFF);
+      AIFlags = PHASE_TURN_START | PHASE_TURN_END;
+    }
+
+    @Override
+    protected void adjustCost()
+    {
+      // Override default cost-increase behavior to make this get more expensive faster.
+      numActivations++;
+      myPowerCost = 1 + (numActivations*3);
+    }
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      // Increase Ave's sphere of influence.
+      Ave.MAX_SNOW_SPREAD_RANGE++;
+
+      // Buff units.
+      damageMod.apply(Ave);
+
+      // Drop snow everywhere inside her range.
+      ArrayList<MapChangeEvent.EnvironmentAssignment> snowTiles = new ArrayList<MapChangeEvent.EnvironmentAssignment>();
+      HashSet<XYCoord> tiles = Utils.findLocationsNearProperties(gameMap, Ave, Ave.MAX_SNOW_SPREAD_RANGE);
+      for( XYCoord coord : tiles )
+      {
+        if( Ave.snowMap[coord.xCoord][coord.yCoord] < CommanderAve.SNOW_THRESHOLD )
+        {
+          Ave.snowMap[coord.xCoord][coord.yCoord] = CommanderAve.SNOW_THRESHOLD;
+          if( gameMap.getEnvironment(coord).weatherType != Weathers.SNOW )
+          {
+            snowTiles.add(new MapChangeEvent.EnvironmentAssignment(coord, Environment.getTile(gameMap.getEnvironment(coord).terrainType, Weathers.SNOW), 1));
+          }
+        }
+      }
+
+      // Do all of our terrain alterations.
+      MapChangeEvent event = new MapChangeEvent(snowTiles);
+      event.performEvent(gameMap);
+      GameEventListener.publishEvent(event);
+    }
+  }
+
+  /**
+   *  Boosts the snow-aura around her buildings, and allows it to (briefly) extend further than normal.
+   *  Deposits snow in a three-space range around all of her units.
+   *  Enemies within 2 spaces of one of Ave's units or buildings are frozen for one turn.
+   */
+  private static class GlacioAbility extends CommanderAbility
+  {
+    private static final String GLACIO_NAME = "Glacio";
+    private static final int GLACIO_COST = 6;
+    private static final int GLACIO_BUFF = 10; // Standard 10
+    private static final int GLACIO_SNOW_SPREAD = 3;
+    private static final int GLACIO_FREEZE_RANGE = 2;
+
+    CommanderAve Ave;
+    COModifier damageMod = null;
+
+    GlacioAbility(CommanderAve commander)
+    {
+      super(commander, GLACIO_NAME, GLACIO_COST);
+      Ave = commander;
+      damageMod = new CODamageModifier(GLACIO_BUFF);
+      AIFlags = PHASE_TURN_START | PHASE_TURN_END;
```

Comment:
I think I'd recommend Glacio be turn end only, while Oblido can stay both.
Glacio doesn't produce all that much benefit on your turn, but it's almost Lightning Strike if you use it at the end of your turn.
Oblido can help you get kills during your turn, though, and Oblido having an extra use phase means the AI might choose one or the other, with them at equal costs.

Reply:
Using Oblido at turn end means you don't get to use your extra firepower though. It does hurt nearby enemies, which might slow their roll next turn, but I'm not sure if that's compelling enough not to just make it turn-begin only.
also Oblido and Glacio are currently not the same cost. I made Glacio cheaper since it doesn't do damage. Think they should be at parity?

Reply:
> Using Oblido at turn end means you don't get to use your extra firepower though.

20 attack is nice, but not exactly game-ending. It's literally Strong's passive, haha.

> It does hurt nearby enemies, which might slow their roll next turn, but I'm not sure if that's compelling enough not to just make it turn-begin only.

2 HP damage, reaching into the enemy formation, is pretty good, especially if you just commit an inf to stare down some tanks (like Muriel seems to enjoy).
Also, powers are supposed to be "ice breakers", so you likely won't have directs committed to get value out of it by the time you use it.

> also Oblido and Glacio are currently not the same cost. I made Glacio cheaper since it doesn't do damage. Think they should be at parity?

As much as I hate mass damage... look at it this way: Olaf and Eagle are considered pretty balanced with each other, and both are mostly bland D2D.
Olaf produces crippling weather and 2 mass damage for 7 stars.
Eagle gets a second turn for non-infantry for *9*.
Neither of Ave's abilities really match up to their global effects (and it's fair to note that our faster charging and Ave's D2D should factor into the costs as well), but we can draw some parallels here.
Oblido is kind of like Olaf's SCOP, but without any of that nasty snow.
Glacio is kind of like Lightning Strike, but it *also* has snow. Not only that, but it works just as well for infantry as it does for anything else - you might very well get to simulate Victory March by freezing your enemy's own units in position to protect your capping infantry.

Especially if we go through with my sinister plans to make mass damage charge COP meters, I think Glacio is perhaps deserving of the greater cost.

Reply:
> 20 attack is nice, but not exactly game-ending. It's literally Strong's passive, haha.

So you are saying I should bump it to 30? Also, I think I want to nerf Strong's passive at some point, in exchange for some more unique features.

> 2 HP damage, reaching into the enemy formation, is pretty good, especially if you just commit an inf to stare down some tanks 

Good point. Worth leaving as both, if just to see how it plays.

> blah blah ... I think Glacio is perhaps deserving of the greater cost.

I would hesitate to draw such a direct parallel between Glacio and Lightning Strike - Glacio doesn't affect as much. Also there isn't any precedent for deforestation, which skews any estimates on Oblido as well. Given the reduced scope compared to Olaf, I could see reducing the cost though. Maybe just move them both to 7? Or do you strongly feel that Glacio is worth more/Oblido less?

Reply:
> > 20 attack is nice, but not exactly game-ending. It's literally Strong's passive, haha.
> 
> So you are saying I should bump it to 30? Also, I think I want to nerf Strong's passive at some point, in exchange for some more unique features.

No especially big opinion there.
Nerfing Strong sounds good. Assuming we change the dynamics around infantry enough, we might even be able to leave the extra infantry production in.

> > blah blah ... I think Glacio is perhaps deserving of the greater cost.
> 
> I would hesitate to draw such a direct parallel between Glacio and Lightning Strike - Glacio doesn't affect as much. Also there isn't any precedent for deforestation, which skews any estimates on Oblido as well. Given the reduced scope compared to Olaf, I could see reducing the cost though. Maybe just move them both to 7? Or do you strongly feel that Glacio is worth more/Oblido less?

Glacio affects as much as Oblido does, so it's as comparable to LS as Oblido is to Olaf SCOP.
The deforestation is something I hadn't really taken into account, though.

I feel like keeping them both at 8 sounds about right, TBH. The caveat I'd add here is that Oblido shouldn't scale cost as quickly as Glacio - possibly not at all. Deforestation is very powerful, but you can only remove a forest once, haha.

Reply:
Or maybe just swap them so that they won't be the same cost? I feel like any cost we assign is probably up for debate, so I don't really have strong feelings on this. Somewhere in this ballpark, anyway.

Reply:
> > Oblido shouldn't scale cost as quickly as Glacio - possibly not at all. Deforestation is very powerful, but you can only remove a forest once, haha.
> 
> Maybe just make Oblido cheaper then? I don't think it should _not_ scale, but starting at 6 instead of 8 might be reasonable then, I dunno.

My reasoning here is that Oblido *should* have a high entrance fee: it makes the map much more favorable for Ave, at the direct cost of her opponents.
At the same time, it's not quite as useful after the first cast, which is why I suggest slower scaling.

I also think both powers should be pretty expensive, as her passive is extremely strong. Wally, or even Spender, has a pretty good shot against Strong/Muriel (which is certainly somewhat due to Muriel's lack of indirects, but hey).